src/pkg/task/dao/execution.go

```diff
@@ -17,16 +17,39 @@ package dao
 import (
 	"context"
 	"fmt"
+	"strconv"
 	"strings"
 	"time"
 
 	"github.com/goharbor/harbor/src/jobservice/job"
+	"github.com/goharbor/harbor/src/jobservice/logger"
+	"github.com/goharbor/harbor/src/lib/cache"
+	"github.com/goharbor/harbor/src/lib/config"
 	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/gtask"
 	"github.com/goharbor/harbor/src/lib/log"
 	"github.com/goharbor/harbor/src/lib/orm"
 	"github.com/goharbor/harbor/src/lib/q"
 )
 
+func init() {
+	// register the execution status refresh task if enable the async update
+	if interval := config.GetExecutionStatusRefreshIntervalSeconds(); interval > 0 {
+		gtask.DefaultPool().AddTask(NewExecutionDAO().(*executionDAO).scanAndRefreshOutdateStatus, time.Duration(interval)*time.Second)
```

Comment:
`NewExecutionDAO().(*executionDAO).scanAndRefreshOutdateStatus` can be simplify to a function, because NewExecutionDAO() is a function need to call at init, there is a risk of adding some cycle dependency related to init in future.